### DISCUSSION TITLE
How to prevent session expirity

### DISCUSSION THREAD

**Discussion by b4d4** [directus/directus#10654] [Archive]
HI, i have a app where i need to login one time and make session infinite. But not directus session expires every 15 min, and i need to restart the password in order to login again.

Thanks

**Comment by benhaynes** [comment]
Did you look into static access tokens?

https://docs.directus.io/reference/authentication/#access-tokens

**Reply by b4d4** [reply]
Thanks for the reply. Can i increase the JWT expiration time?

**Comment by b4d4** [comment]
This problem also apperas in directus dashboard. The sessión expires and i cant login, i need to reset to password in order can login in directus.

**Reply by Pachat** [reply]
May be related to #10726 [_see comment_](https://github.com/directus/directus/discussions/10726#discussioncomment-1884916) in case your have multiple sessions running on your client

**Comment by rijkvanzanten** [comment]
> HI, i have a app where i need to login one time and make session infinite. But not directus session expires every 15 min, and i need to restart the password in order to login again.

I'm not sure what you mean by _restart the password_. I'm assuming you mean login again with email/password.

What you're looking for is the refresh token. The access token is very short-lived, as it's sent back and forth to the API. Once it expires, you can use the `/auth/refresh` endpoint to get a new access token using your refresh token.

**Reply by b4d4** [reply]
Hi, i updated to the latest directus verison (9.4.3) and the error of session expiration still here.
The scenario is as follows. I am logged into the application and an app of my own. When some time passes and I try to log into my own app, the login fails. Attached is the screenshot. And if I try to log into directus, it doesn't let me either. Same failure. I don't know what to try anymore. 

![image](https://user-images.githubusercontent.com/58529478/148978935-7c88aeb2-0723-4d7c-a7cc-ba1314afba69.png)

**Reply by rijkvanzanten** [reply]
Do you have any browser extensions or proxy servers that might prevent the cookie header from being set / read correctly?

**Comment by b4d4** [comment]
I tried with other browser and same error. It cant make the refresh token
request. (The  dirty solution, is to request new password in order to reset
token...)

**Reply by rijkvanzanten** [reply]
> It cant make the refresh token
request.

Any errors?

**Comment by b4d4** [comment]
Yes, INVALID CREDENTIALS. Error 401.

**Reply by rijkvanzanten** [reply]
Do you have any browser extensions or proxy servers that might prevent the cookie header from being set / read correctly?

**Comment by b4d4** [comment]
I tried with other browsers without any extension.

**Comment by halwesit** [comment]
add this to your `.env`
`
ACCESS_TOKEN_TTL=1440m
`
this makes the token expires after 24 hours, you can change the 1440m to increase or decrease the expiration time.

> reference: https://docs.directus.io/configuration/config-options/#security

**Comment by b4d4** [comment]
Hello, after a long time we are developing another application using the directus api. In the app we use the methods to authenticate, refresh the token, logout...etc. After x amount of time, we try to log in and it tells us incorrect credentials. I have entered the database and it changes the hash of the password for all users. Is there any process that alters the passwords?

**Reply by rijkvanzanten** [reply]
No passwords aren't updated automatically.

**Comment by b4d4** [comment]
After a long time have found the solution to my problem, and it is, that when I make a request putting in the url (*.*.* ...etc) it also brought me ALL the user data INCLUDING the password but with asterisks ******. Which when I send the whole object, it also sends me the password and changes it for asterisks *******

I know that I must bring me only the fields that I want to modify and not to bring me all the levels with asterisks ***.

As a solution for the directus team, I would recommend that the password field of the users table is not brought in the calls.

**Reply by minimorcy** [reply]
Hi! There is something you can do. You can limit, per role, which fields you send in the 'get' calls. So you can limit the 'get' calls to '/users' to not bring you the password field. For your "problem" it could be enough, since you would receive the user object without the password field, avoiding it to be updated if you pass it the whole object.
